packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -99,17 +105,20 @@ export class Outbox {
 	private maybeFlushPartialBatch() {
 		const mainBatchSeqNums = this.mainBatch.sequenceNumbers;
 		const attachFlowBatchSeqNums = this.attachFlowBatch.sequenceNumbers;
+		const blobAttachSeqNums = this.blobAttachBatch.sequenceNumbers;
 		assert(
 			this.params.config.disablePartialFlush ||
-				sequenceNumbersMatch(mainBatchSeqNums, attachFlowBatchSeqNums),
+				(sequenceNumbersMatch(mainBatchSeqNums, attachFlowBatchSeqNums) &&
+					sequenceNumbersMatch(mainBatchSeqNums, blobAttachSeqNums)),
 			0x58d /* Reference sequence numbers from both batches must be in sync */,
 		);
 
 		const currentSequenceNumbers = this.params.getCurrentSequenceNumbers();
 
 		if (
 			sequenceNumbersMatch(mainBatchSeqNums, currentSequenceNumbers) &&
-			sequenceNumbersMatch(attachFlowBatchSeqNums, currentSequenceNumbers)
+			sequenceNumbersMatch(attachFlowBatchSeqNums, currentSequenceNumbers) &&
```

Comment:
@andre4i - FYI, this is related to our discussion of what is the "op/base reference" in the new world.
I think one of the biggest questions - what are other places that use (ref)sequenceNumber to define a base (for merges / represent states of container), and what is the semantic of it. I think it should be - state of the container after processing a batch (similar to how it was before - after processing an op).
That said, I really worry about reentrancy - to some extent any code that touched any properties related to "current" sequence number while we are in the middle of processing a batch is a suspect. I think long term we have to move away from our current eventing model - I think we need to only raise one event for whole container at the end of processing a set of ops (where set is at least a batch, but could be larger), with tools to decompose this event into more granular "this region is invalidated" kind of events for individual DDSs, but no ordering assumptions.
Meanwhile, we need to ensure that we could move forward with this proposal with some set of safety net validations. I'm not sure what they are, to be honest. Testing that product does not break seems insufficient.